VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -180,8 +180,16 @@ public void onRestart() {
             mSelectedItem = -1;
         }
 
-        nextPage = 1;
-        controller.resetSilently();
+        /* Dev Notes:
+        If the activity/fragment needs to be reinstantiated upon restoration,
+        then in some cases the onRestart() callback maybe invoked before view
+        initialization, and thus the controller might not be initialized, and
+        therefore we need to guard this with a null check.
+         */
+        if (controller != null) {
+            nextPage = 1;
```

Comment:
right, but don't we only need to reset the controller if the view is getting reloaded? Otherwise we can just show existing data? Otherwise, we'll be nuking the user's context?